### PR TITLE
JOI validation

### PR DESCRIPTION
**Pull request by JIdayyy**

### CODE REVIEW

File: src/api/users/interfaces.ts [hunk 1]
```diff
@@ -43,7 +44,7 @@ interface Params {
 export default interface UserHandlers {
   getAll: RequestHandler<null, UserWithoutPassword[], null>;
   getOne: RequestHandler<Params, UserWithoutPassword, null>;
-  post: RequestHandler<null, UserWithoutPassword, ReqBodyUserPost>;
-  put: RequestHandler<Params, null, ReqBodyUserPut>;
+  post: RequestHandler<any, UserWithoutPassword | Error, ReqBodyUserPost>;
+  put: RequestHandler<any, null, ReqBodyUserPut>;
```
Comment (mryckewaert) [inline src/api/users/interfaces.ts]: add params and null

**Review by mryckewaert — COMMENTED** [review]
We need to add a params

**Review by mryckewaert — APPROVED** [review]

**Review by Thiez-64 — APPROVED** [review]
ok

**Review by mryckewaert — APPROVED** [review]

**Review by Tweaka — APPROVED** [review]

**Review by Thiez-64 — APPROVED** [review]
ok

**Pull request merged**